### DISCUSSION TITLE
WSL Support

### DISCUSSION THREAD

**Discussion by timurkh-devstark** [modelcontextprotocol/.github#29] [Q&A] [answered]
Hi! Is it possible to add a WSL support? Not sure where should I send this request. Currently I've tried to create a simple weather-app project in WSL, Windows 11, Claude Desktop app.

Currently I have mcpServer config like this:
```
"command": "node",
            "args":
            [
                "\\wsl.localhost\\Ubuntu-22\\home\\timur\\...\\build\\index.js"
            ],
```

mcp server log:
```
Error: Cannot find module 'C:\wsl.localhost\Ubuntu-22\home\timur\...\build\index.js'
```

Seems like it tries to access wsl.localhost at C: drive

**Comment by dsp-ant** [comment, marked as answer]
I know little about how to use WSL. https://learn.microsoft.com/en-us/windows/wsl/filesystems seems to suggest that

```
            "command": "wsl",
            "args":
            [
                "node",
                "path/to/index.js"
            ],
```

Alternatively: 
```
            "command": "node",
            "args":
            [
                "\\wsl$\<distroname>\<path>"
            ],
 ```
            
might work, assuming node is installed on windows.

/cc @anaisbetts in case you know how WSL works on Windows.

**Reply by timurkh-devstark** [reply]
Yep, thank you, that's work!

**Comment by cmosguy** [comment]
Unfortunately, this does not work, is there some port that gets blocked between WSL and claud desktop?
